src/test/java/org/manifold/compiler/front/TestPrimitiveNodeVertex.java

```diff
@@ -0,0 +1,372 @@
+package org.manifold.compiler.front;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.log4j.ConsoleAppender;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.PatternLayout;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.manifold.compiler.BooleanTypeValue;
+import org.manifold.compiler.NilTypeValue;
+import org.manifold.compiler.NodeTypeValue;
+import org.manifold.compiler.TypeValue;
+import org.manifold.compiler.Value;
+
+public class TestPrimitiveNodeVertex {
```

Comment:
Methods annotated with `@Before` executes before each test, and can be used to facilitate some of the setup code (initialize as member variables). Probably not worth refactoring all of this, something for the future to make individual test cases shorter :)